app/reflexes/products_reflex.rb

```diff
@@ -3,36 +3,52 @@
 class ProductsReflex < ApplicationReflex
   include Pagy::Backend
 
-  def fetch
-    @page ||= element.dataset.page || 1
-    @per_page ||= element.dataset.perpage || 15
-    @search_term ||= element.dataset.searchterm || ""
-
-    fetch_products
+  before_reflex :init_params
 
-    render_products
+  def fetch
+    fetch_and_render_products
   end
 
   def change_per_page
     @per_page = element.value.to_i
     @page = 1
 
-    fetch
+    fetch_and_render_products
   end
 
   def filter
-    @per_page = params[:per_page]
     @page = 1
-    @search_term = params[:search_term]
-    @producer_id = params[:producer_id]
-    @category_id = params[:category_id]
 
-    fetch_products
-    render_products
+    fetch_and_render_products
   end
 
   private
 
+  def init_params
+    init_filters_params
+    init_pagination_params
+  end
```

Comment:
Why not just hook these up directly with `before_reflex`?

Reply:
Yes, that's probably easier to understand ;)